SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -873,15 +877,19 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 	if (SpatialGDKSettings->bAsyncLoadNewClassesOnEntityCheckout && NeedToLoadClass(ClassPath))
 	{
 		StartAsyncLoadingClass(ClassPath, EntityId);
-		return;
+		return nullptr;
 	}
 
 	AActor* EntityActor = Cast<AActor>(PackageMap->GetObjectFromEntityId(EntityId));
 	if (EntityActor != nullptr)
 	{
-		UE_LOG(LogSpatialReceiver, Verbose, TEXT("%s: Entity %lld for Actor %s has been checked out on the worker which spawned it."),
-			   *NetDriver->Connection->GetWorkerId(), EntityId, *EntityActor->GetName());
-		return;
+		if (!EntityActor->IsActorReady())
```

Comment:
Can we avoid changing ReceiveActor if we check if the actor is ready in OnAddComponent when the metadata is added?

Reply:
If the actor is not added to the pending actors, there is no opportunity to go through the path that drops the updates (or maybe https://improbableio.atlassian.net/browse/UNR-3457 deals with it? but that is advertised as a workaround to be removed).
I would have to add the check for every component received, not just the metadata. Maybe it is cleaner ? But it would change the current flow more.